File: JS/UberBlack.js
var {Car} = require("./Car");

function UberBlack(license, driver , typeCarAccepted, seatsMaterial) {
    this.typeCarAccepted = typeCarAccepted;
    this.seatsMaterial = seatsMaterial;
    Car.call(this,license,driver);
}

UberBlack.prototype = new Car();

UberBlack.prototype.getDataCar = function() {
    if (Car.prototype.getDataCar.call(this)!="Error!" && this.typeCarAccepted!=undefined && this.seatsMaterial!=undefined){
        return Car.prototype.getDataCar.call(this) + ", Type-Car-Accepted: " + this.typeCarAccepted + ", Seat-Material: " + this.seatsMaterial;
    }
    return "Error!";
}

module.exports = {
    UberBlack: UberBlack
}

File: Java/UberBlack.java
import java.util.ArrayList;
import java.util.Map;

public class UberBlack extends Car{
    Map<String, Map<String,Integer>> typeCarAccepted;
    // Marca        Modelo, Año
    ArrayList<String> seatsMaterial;
    //String[] seatsMaterial = {"Vinil","Piel"};

    public UberBlack(String license, Account driver, Map<String, Map<String,Integer>> typeCarAccepted, ArrayList<String> seatsMaterial) {
        super(license,driver);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
    }

    @Override
    public String getDataCar() {
        if (typeCarAccepted!=null && seatsMaterial!=null) {
            return super.getDataCar() + ", Type-Car-Accepted: " + typeCarAccepted + ", Seat-Material: "+ seatsMaterial;
        }
        return "Error!";
    }
}
